finch/processes/wps_sdba.py

```diff
@@ -0,0 +1,220 @@
+                ),
+                LiteralInput(
+                    "kind",
+                    "Kind of adjustment (+, *)",
+                    abstract="Use * for multiplicative adjustment, or + for additive adjustement.",
+                    data_type="string",
+                    default="+",
+                    allowed_values=["+", "*"],
+                    min_occurs=0,
+                ),
+            ]
+        )
+
+        super().__init__(
+            self._handler,
+            identifier="empirical_quantile_mapping",
+            title="Empirical Quantile Mapping bias-adjustment",
+            version="0.1",
+            abstract="Adjustment factors are computed between the quantiles of `ref` and `sim`."
+            "Values of `sim` are matched to the corresponding quantiles of `hist` and corrected accordingly.",
+            inputs=inputs,
+            outputs=common_outputs,
+            status_supported=True,
+            store_supported=True,
+        )
+
+    def _handler(self, request, response):
+        def _log(message, percentage):
+            write_log(self, message, subtask_percentage=percentage)
+
+        res = {}
+        group = {}
+        train = {}
+        adj = {}
+
+        try:
+            variable = request.inputs.pop(wpsio.variable.identifier, None)
+
+            for key, input in request.inputs.items():
+                if key in resources:
+                    ds = try_opendap(request.inputs[key][0])
+                    name = variable or list(ds.data_vars)[0]
+                    res[key] = ds[name]
+
+                elif key in group_args:
+                    group[key] = single_input_or_none(request.inputs, key)
+
+                elif key in adjust_args:
+                    adj[key] = single_input_or_none(request.inputs, key)
+
+                else:
+                    train[key] = single_input_or_none(request.inputs, key)
+
+            _log("Successfully read inputs from request.", 1)
+
+            group = xclim.sdba.Grouper(**group)
+            _log("Grouper object created.", 2)
+
+            bc = xclim.sdba.EmpiricalQuantileMapping(**train, group=group)
+            bc.train(res["ref"], res["hist"])
+            _log("Training object created.", 3)
+
+            out = bc.adjust(res["sim"], interp=group["interp"]).to_dataset(name=name)
+            _log("Adjustment object created.", 5)
+
+            out_fn = Path(self.workdir) / "bias_corrected.nc"
+            with FinchProgressBar(
+                logging_function=_log,
+                start_percentage=5,
+                end_percentage=98,
+                width=15,
+                dt=1,
+            ):
+                dataset_to_netcdf(out, out_fn)
+
+        except Exception as exc:
+            LOGGER.exception(exc)
+            err_msg = traceback.format_exc()
+            raise ProcessError(
+                err_msg, max_length=len(err_msg), allowed_chars=string.printable
+            ) from exc
```

Comment:
This is already done in `FinchProcess._handler_wrapper`. But here you have a bit stuff you could move there.